`build_qt.py` fails with Xcode 26.4 on macOS

## Description
Due to [QTBUG-145239](https://qt-project.atlassian.net/browse/QTBUG-145239), `build_qt.py` fails with Xcode 26.4 as follows:
```
In file included from /build/qtbase/include/QtCore/qyieldcpu.h:1:
/src/qtbase/src/corelib/thread/qyieldcpu.h:37:5: error: implicitly declaring library function '__yield' with type 'void ()'
[-Werror,-Wimplicit-function-declaration]
   37 |     __yield();              // Generic
      |     ^
/src/qtbase/src/corelib/thread/qyieldcpu.h:37:5: note: include the header <arm_acle.h> or explicitly provide a declaration for '__yield'
```

## Steps to reproduce
1. Install Xcode 26.4
2. `python3 build_tools/build_qt.py --release --confirm_license`

## Expected behavior
* The build succeeds without any error at the step 2.

## Actual behavior
* The build fails due to [QTBUG-145239](https://qt-project.atlassian.net/browse/QTBUG-145239)

## Version or commit-id
395f9d7fa9c90e265c953a6deaea2922a368a12a

## Environment
 - OS: macOS 26.4
 - Xcode 26.4

## Additional context
According to [QTBUG-145239](https://qt-project.atlassian.net/browse/QTBUG-145239), Qt 6.11.1 and later should include the fix. Qt 6.11.1 is expected to be released [on April 30, 2026](https://wiki.qt.io/Qt_6.11_Release#Qt_6.11_release_plan) as of writing.

One tricky aspect is that Qt 6.10 and later bumps minimum supported macOS version from macOS 12 to macOS 13, compared to Qt 6.9.1, which we are currently using.

--- a/src/build_tools/build_qt.py
+++ b/src/build_tools/build_qt.py
@@ -840,6 +840,36 @@ def extract_qt_src(args: argparse.Namespace) -> None:
           f.extractall(path=qt_src_dir, filter=filter)
       else:
         f.extractall(path=qt_src_dir, members=qt_extract_filter(f))
+  # TODO: https://github.com/google/mozc/issues/1457 - Remove this.
+  apply_patch_to_qt_src(qt_src_dir, args.dryrun)
+
+
+def apply_patch_to_qt_src(
+    qt_src_dir: pathlib.Path,
+    dryrun: bool = False,
+) -> None:
+  """Apply local patches to the extracted Qt source.
+
+  Args:
+    qt_src_dir: The root directory of the extracted Qt source.
+    dryrun: True to skip actual file modifications.
+  """
+  # Workaround for QTBUG-145239: qyieldcpu.h fails to compile with
+  # macOS 26.4 SDK on Apple Silicon.
+  # https://qt-project.atlassian.net/browse/QTBUG-145239
+  # https://codereview.qt-project.org/c/qt/qtbase/+/724619
+  qyieldcpu_h = qt_src_dir.joinpath('src', 'corelib', 'thread', 'qyieldcpu.h')
+  if dryrun:
+    print(f'dryrun: patching {qyieldcpu_h} for QTBUG-145239')
+    return
+  content = qyieldcpu_h.read_text(encoding='utf-8')
+  content = content.replace(
+      '#if __has_builtin(__yield)\n',
+      '#if __has_builtin(__builtin_arm_yield)\n'
+      '    __builtin_arm_yield();\n'
+      '#elif __has_builtin(__yield)\n',
+  )
+  qyieldcpu_h.write_text(content, encoding='utf-8')
 
 
 def main():

